src/System.Globalization.Calendars/tests/CalendarHelpers.cs

```diff
@@ -0,0 +1,185 @@
+        {
+            return calendar.GetEra(calendar.MaxSupportedDateTime);
+        }
+
+        private static int YearOffset(Calendar calendar)
+        {
+            if (calendar is KoreanCalendar)
+            {
+                return 2333;
+            }
+            if (calendar is ThaiBuddhistCalendar)
+            {
+                return 543;
+            }
+            return 0;
+        }
+
+        public static void AddYears_Invalid(Calendar calendar)
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => calendar.AddYears(calendar.MaxSupportedDateTime, 1));
+            Assert.Throws<ArgumentOutOfRangeException>(() => calendar.AddYears(calendar.MinSupportedDateTime, -1));
+        }
+
+        public static void AddMonths_Invalid(Calendar calendar)
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => calendar.AddMonths(calendar.MaxSupportedDateTime, 1));
+            Assert.Throws<ArgumentOutOfRangeException>(() => calendar.AddMonths(calendar.MinSupportedDateTime, -1));
+
+            Assert.Throws<ArgumentOutOfRangeException>("months", () => calendar.AddMonths(DateTime.Now, -120001));
+            Assert.Throws<ArgumentOutOfRangeException>("months", () => calendar.AddMonths(DateTime.Now, 120001));
+        }
+
+        public static void GetDaysInYear_Invalid(Calendar calendar)
+        {
+            int yearOffset = YearOffset(calendar);
+            int era = MaxEra(calendar);
+
+            // Year is outside supported range
+            Assert.Throws<ArgumentOutOfRangeException>("year", () => calendar.GetDaysInYear(-1, era));
+            Assert.Throws<ArgumentOutOfRangeException>("year", () => calendar.GetDaysInYear(calendar.MinSupportedDateTime.Year + yearOffset - 1, era));
+            Assert.Throws<ArgumentOutOfRangeException>("year", () => calendar.GetDaysInYear(calendar.MaxSupportedDateTime.Year + yearOffset + 1, era));
+
+            // Era is outside supported range
+            Assert.Throws<ArgumentOutOfRangeException>("era", () => calendar.GetDaysInYear(yearOffset + 1, -era));
+            Assert.Throws<ArgumentOutOfRangeException>("era", () => calendar.GetDaysInYear(yearOffset + 1, era + 1));
+        }
+
+        public static void GetDaysInMonth_Invalid(Calendar calendar)
+        {
+            int yearOffset = YearOffset(calendar);
+            int era = MaxEra(calendar);
+
+            // Year is outside supported range
+            Assert.Throws<ArgumentOutOfRangeException>("year", () => calendar.GetDaysInMonth(-1, 10, era));
+            Assert.Throws<ArgumentOutOfRangeException>("year", () => calendar.GetDaysInMonth(0, 10, era));
+            Assert.Throws<ArgumentOutOfRangeException>("year", () => calendar.GetDaysInMonth(calendar.MinSupportedDateTime.Year + yearOffset - 1, 10, era));
+            Assert.Throws<ArgumentOutOfRangeException>("year", () => calendar.GetDaysInMonth(calendar.MaxSupportedDateTime.Year + yearOffset + 1, 10, era));
```

Comment:
Good point. What I can do, is at some point after this PR I can modify CalendarHelpers to support each era